benchmark/TensorKitBenchmarks/utils/BenchUtils.jl

```diff
@@ -0,0 +1,74 @@
+module BenchUtils
+
+export parse_type, expand_kwargs
+export generate_space
+
+using TensorKit
+
+parse_type(x::String) = eval(Meta.parse(x))
+
+function expand_kwargs(params::Dict)
+    const_params = NamedTuple(Symbol(key) => value
+                              for (key, value) in params if !(value isa Vector))
+    nonconst_keys = Tuple(Symbol(key) for (key, value) in params if value isa Vector)
+    nonconst_vals = (value for value in values(params) if value isa Vector)
+    return Iterators.map(Iterators.product(nonconst_vals...)) do expanded_vals
+        return merge(const_params, NamedTuple{nonconst_keys}(expanded_vals))
+    end
+end
+
+"""
+    generate_space(::Type{I}, D::Int; sigma::Real=1.0)
+
+Creates a (graded) vector space with sectortype `I` and total dimension `D`, where the distribution of charges is controlled through a spread parameter `sigma`.
+"""
+function generate_space(::Type{Trivial}, D::Int, sigma::Nothing=nothing)
+    return ComplexSpace(round(Int, D))
+end
+function generate_space(::Type{Z2Irrep}, D::Int, sigma::Real=0.5)
+    D_even = ceil(Int, sigma * D)
+    D_odd = D - D_even
+    return Z2Space(0 => D_even, 1 => D_odd)
+end
+function generate_space(::Type{U1Irrep}, D::Int, sigma::Real=0.5)
+    # use ceil here to avoid getting stuck
+    normal_pdf = let D = D
+        x -> ceil(Int, D * exp(-0.5 * (x / sigma)^2) / (sigma * sqrt(2π)))
+    end
+
+    sectors = U1Irrep[]
+    dims = Int[]
+
+    for sector in values(U1Irrep)
+        round(sector.charge) == sector.charge || continue
+        d = normal_pdf(sector.charge)
+        push!(sectors, sector)
+        push!(dims, d)
+        D -= d
+        D < 1 && break
+
+        if abs(sector.charge) > 20
+            error()
+        end
+    end
+
+    return U1Space((s => d for (s, d) in zip(sectors, dims))...)
+end
+function generate_space(::Type{SU2Irrep}, D::Int, sigma::Real=0.5)
+    poisson_pdf(x) = ceil(Int, D * exp(-sigma) * sigma^x / factorial(x))
```

Comment:
Strictly speaking, this is not correctly normalised. If we assume that `SU2Irrep(j)` is indeed generated with dimension `poisson_pdf(2*j) = D * exp(-sigma) * sigma^(2j) / factorial(2j)`, then the total dimension is, with `n=2j`

$$\sum_{n=0}^{+\infty} D * (n+1) * exp(-\sigma) * \sigma^n / n! \neq D$$

due to the extra factor $$n+1=2j+1$$, which comes from the dimension of the irrep.

Maybe `poisson_pdf(x) = ceil(Int, D * exp(-sigma) * sigma^x / factorial(x) / (x+1))`

which happens to be

Maybe `poisson_pdf(x) = ceil(Int, D * exp(-sigma) * sigma^x / factorial(x+1))`

as alternative

Reply:
Also, for large sigma, the poisson distribution is peaked away from zero, so I do not know if that is really the distribution we want, versus just a half gaussian, but I guess those details don't really matter (but we probably do not want to change them afterwards).

Reply:
I'm definitely happy to switch that, although I think the peak is reasonable too: I like poisson for SU(2) since in the Heisenberg virtual spaces often there are more values of spin 1 than spin 0.

Probably it doesn't matter too much for the actual representativeness of the benchmarks though